component/block_cache/block_cache.go

```diff
@@ -253,6 +254,23 @@ func (bc *BlockCache) Configure(_ bool) error {
 		bc.diskTimeout = conf.DiskTimeout
 	}
 
+	var noDu bool
+	if err := config.UnmarshalKey("no-du", &noDu); err == nil {
+		if noDu {
+			bc.diskUsageConfiguration = common.DiskUsageConfiguration{
+				DiskUsageFunction: common.GetUsageWithWalkInMegabytes,
+				UsesDu:            false,
```

Comment:
If the callback method is set then do we need this flag marking use du or no du. Based on the method linked here in function pointer we have any way made a decision to use or not use du.